saleor/checkout/complete_checkout.py

```diff
@@ -1598,17 +1598,16 @@ def create_order_from_checkout(
 
     code = None
 
-    if voucher := checkout_info.voucher:
-        with transaction.atomic():
-            code = _increase_voucher_code_usage_value(checkout_info=checkout_info)
-
     with transaction.atomic():
         checkout_pk = checkout_info.checkout.pk
         checkout = Checkout.objects.select_for_update().filter(pk=checkout_pk).first()
         if not checkout:
             order = Order.objects.get_by_checkout_token(checkout_pk)
             return order
 
+        if voucher := checkout_info.voucher:
```

Comment:
This was placed outside of the atomic on purpose. Single Voucher/code, can be used in multiple checkouts. Calling the `increase_voucher_usage` here, means, that we will fully block usage of this voucher for the time of this transaction. This will slow down the rest of the checkouts with the same voucher: you lock usage of voucher for next  +/- 1-2 sec (as this is the time of `_create_order_from_checkout`). 
Like:
- Checkout A is in line 1620. voucher is locked
- Checkout B tries to apply the lock on the same code. It waits for Checkout A
- Checkout C tries to apply the  lock on the same code. It waits for Checkout A
- Checkout D tries to apply the  lock on the same code. It waits for Checkout A


If possible, I would suggest to keep the previous version

Reply:
will change that